src/rpc/file_replication_handler.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright 2025 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include "rpc/file_replication_handler.hpp"
+#include "slk/streams.hpp"
+#include "storage/v2/durability/paths.hpp"
+#include "storage/v2/replication/serialization.hpp"
+#include "utils/logging.hpp"
+
+namespace memgraph::rpc {
+
+FileReplicationHandler::~FileReplicationHandler() { ResetCurrentFile(); }
+
+std::filesystem::path FileReplicationHandler::GetRandomDir() {
+  auto const random_str = utils::GenerateUUID();
+  return std::filesystem::temp_directory_path() / "memgraph" / random_str /
+         storage::durability::kReplicaDurabilityDirectory;
+}
+
+// The assumption is that the header, request, file name and file size will always fit into the buffer size = 64KiB
+// Currently, they are taking few hundred bytes at most so this should be a valid assumption. Also, we aren't expecting
+// big growth in message size/
+std::optional<size_t> FileReplicationHandler::OpenFile(const uint8_t *data, size_t const size) {
+  auto const tmp_rnd_dir = GetRandomDir();
+
+  if (!utils::EnsureDir(tmp_rnd_dir)) {
+    spdlog::error("Failed to create temporary directory {}", tmp_rnd_dir);
+    return std::nullopt;
+  }
+
+  slk::Reader req_reader(data, size, size);
+  storage::replication::Decoder decoder(&req_reader);
+
+  auto const maybe_filename = decoder.ReadString();
+  MG_ASSERT(maybe_filename.has_value(), "Filename missing for the received file over the RPC");
+
+  if (maybe_filename->empty()) {
+    spdlog::error("Filename is empty");
+    return std::nullopt;
+  }
+  auto const file_path = std::filesystem::path(*maybe_filename);
+  if (file_path.has_parent_path()) {
+    spdlog::error("File cannot have a parent path{}", file_path.string());
+    return std::nullopt;
+  }
+
+  auto const path = tmp_rnd_dir / file_path;
+  paths_.emplace_back(path);
+
+  file_.Open(path, utils::OutputFile::Mode::OVERWRITE_EXISTING);
```

Comment:
Can we check if it is actually open, maybe some random invalid unicode is not valid on the filesystem.